src/main/java/de/retest/recheck/ui/descriptors/Element.java

```diff
@@ -104,6 +104,14 @@ public Element applyChanges( final ActionChangeSet actionChangeSet ) {
 		return element;
 	}
 
+	/**
+	 * Note that the retestId is immutable, so this method returns a new Element that is a copy of the old one with the
+	 * new retestId.
+	 */
+	public Element setRetestId( final String retestId ) {
+		return Element.create( retestId, this, identifyingAttributes, attributes );
```

Comment:
In most cases `this` isn't `parent` ;)

Reply:
Oh, wow. This looks like a major error: I copied this code from 5 lines above (`public Element applyChanges( final ActionChangeSet actionChangeSet ) {`). Someone needs to look into that.